health/polling_test.go

```diff
@@ -122,10 +122,70 @@ func TestPollingCheckerThresholds(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestPollingCheckerTimeout(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	t.Cleanup(cancel)
+
+	interval := 5 * time.Second
+	testClock := clocktest.NewFakeClock()
+
+	checker := NewPollingChecker(PollingCheckerConfig{
+		PollingInterval: interval,
+		Timeout:         10 * time.Millisecond,
+	}, NewSimpleProber("/"))
+	checker.(*pollingChecker).clock = testClock //nolint:errcheck
+
+	connection := make(fakeConnChan)
+	tracker := make(fakeHealthTracker)
+	process := checker.New(ctx, connection, tracker)
+	advance := func(response *http.Response) {
+		t.Helper()
+		testClock.Advance(interval)
```

Comment:
I couldn't figure out the pattern from the other test and changed it to advance before providing the response - this seemed to match the flow better, trigger the health check and provide the response. In the flow exercised by this test, the other pattern would deadlock since providing the response before triggering a health check would block.